plasmapy_nei/nei/tests/test_nei.py

```diff
@@ -1,6 +1,306 @@
+        "T_e": 1 * u.MK,
+        "n": n_array,
+        "time_input": time_array,
+        "time_start": 0 * u.s,
+        "time_max": 800 * u.s,
+        "dt": 100 * u.s,
+        "max_steps": 2,
+        "adapt_dt": False,
+        "verbose": True,
+    },
+    "n_e constant": {
+        "inputs": inputs_dict,
+        "abundances": abundances,
+        "T_e": T_e_array,
+        "n": 1e9 * u.cm ** -3,
+        "time_input": time_array,
+        "time_start": 0 * u.s,
+        "time_max": 800 * u.s,
+        "max_steps": 2,
+        "adapt_dt": False,
+        "dt": 100 * u.s,
+        "verbose": True,
+    },
+    "T_e function": {
+        "inputs": inputs_dict,
+        "abundances": abundances,
+        "T_e": lambda time: 1e4 * (1 + time / u.s) * u.K,
+        "n": 1e15 * u.cm ** -3,
+        "time_max": 800 * u.s,
+        "max_steps": 2,
+        "dt": 100 * u.s,
+        "adapt_dt": False,
+        "verbose": True,
+    },
+    "n function": {
+        "inputs": inputs_dict,
+        "abundances": abundances,
+        "T_e": 6e4 * u.K,
+        "n": lambda time: 1e9 * (1 + time / u.s) * u.cm ** -3,
+        "time_start": 0 * u.s,
+        "time_max": 800 * u.s,
+        "adapt_dt": False,
+        "dt": 200 * u.s,
+        "verbose": True,
+        "max_steps": 4,
+    },
+    "equil test cool": {
+        "inputs": ["H", "He", "N"],
+        "abundances": {"H": 1, "He": 0.1, "C": 1e-4, "N": 1e-4, "O": 1e-4, "Fe": 1e-4},
+        "T_e": 10001.0 * u.K,
+        "n": 1e13 * u.cm ** -3,
+        "time_max": 2e6 * u.s,
+        "tol": 1e-9,
+        "adapt_dt": False,
+        "dt": 5e5 * u.s,
+        "max_steps": 4,
+        "verbose": True,
+    },
+    "equil test hot": {
+        "inputs": ["H", "He", "C"],
+        "abundances": {
+            "H": 1,
+            "He": 0.1,
+            "C": 1e-4,
+            "N": 1e-4,
+            "O": 1e-4,
+            "Fe": 1e-4,
+            "S": 2e-6,
+        },
+        "T_e": 7e6 * u.K,
+        "n": 1e9 * u.cm ** -3,
+        "time_max": 1e8 * u.s,
+        "dt": 5e7 * u.s,
+        "max_steps": 3,
+        "adapt_dt": False,
+        "verbose": True,
+    },
+    "equil test start far out of equil": {
+        "inputs": {
+            "H": [0.99, 0.01],
+            "He": [0.5, 0.0, 0.5],
+            "O": [0.2, 0, 0.2, 0, 0.2, 0, 0.2, 0, 0.2],
+        },
+        "abundances": {"H": 1, "He": 0.1, "O": 1e-4},
+        "T_e": 3e6 * u.K,
+        "n": 1e9 * u.cm ** -3,
+        "dt": 1e6 * u.s,
+        "time_start": 0 * u.s,
+        "time_max": 1e6 * u.s,
+        "adapt_dt": False,
+        "verbose": True,
+        "max_steps": 2,
+    },
+    "adapt dt": {
+        "inputs": ["H", "He"],
+        "abundances": {"H": 1, "He": 0.1},
+        "T_e": lambda t: u.K * (1e6 + 1.3e4 * np.sin(t.value)),
+        "n": 1e10 * u.cm ** -3,
+        "max_steps": 300,
+        "time_start": 0 * u.s,
+        "time_max": 2 * np.pi * u.s,
+        "adapt_dt": True,
+    },
+}
```

Comment:
I really think this would be better as a fixture, because you're doing some weird stuff below, there... (continued)